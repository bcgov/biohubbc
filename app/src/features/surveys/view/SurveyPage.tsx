import Box from '@mui/material/Box';
import CircularProgress from '@mui/material/CircularProgress';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Paper from '@mui/material/Paper';
import SurveySubmissionAlertBar from 'components/publish/SurveySubmissionAlertBar';
import { SystemRoleGuard } from 'components/security/Guards';
import { SYSTEM_ROLE } from 'constants/roles';
import { CodesContext } from 'contexts/codesContext';
import { ConfigContext } from 'contexts/configContext';
import { SurveyContext } from 'contexts/surveyContext';
import SurveyDetails from 'features/surveys/view/SurveyDetails';
import React, { useContext, useEffect } from 'react';
import SurveyStudyArea from './components/SurveyStudyArea';
import SurveySummaryResults from './summary-results/SurveySummaryResults';
import SurveyObservations from './survey-observations/SurveyObservations';
import SurveyAnimals from './SurveyAnimals';
import SurveyAttachments from './SurveyAttachments';
import SurveyHeader from './SurveyHeader';

//TODO: PRODUCTION_BANDAGE: Remove <SystemRoleGuard validSystemRoles={[SYSTEM_ROLE.DATA_ADMINISTRATOR, SYSTEM_ROLE.SYSTEM_ADMIN]}>

/**
 * Page to display a single Survey.
 *
 * @return {*}
 */
const SurveyPage: React.FC = () => {
  const config = useContext(ConfigContext);
  const codesContext = useContext(CodesContext);
  const surveyContext = useContext(SurveyContext);

  useEffect(() => codesContext.codesDataLoader.load(), [codesContext.codesDataLoader]);

  if (!codesContext.codesDataLoader.data || !surveyContext.surveyDataLoader.data) {
    return <CircularProgress className="pageProgress" size={40} />;
  }

  return (
    <>
      <SurveyHeader />
      <Container maxWidth="xl">
        <Box my={3}>
          <SystemRoleGuard validSystemRoles={[SYSTEM_ROLE.DATA_ADMINISTRATOR, SYSTEM_ROLE.SYSTEM_ADMIN]}>
            <SurveySubmissionAlertBar />
          </SystemRoleGuard>
          <Grid container spacing={3}>
            <Grid item md={12} lg={4}>
              <Paper elevation={0}>
                <SurveyDetails />
              </Paper>
            </Grid>
            <Grid item md={12} lg={8}>
              <Box mb={3}>
                <Paper elevation={0}>
                  <SurveyObservations />
                </Paper>
              </Box>
              <Box mb={3}>
                <Paper elevation={0}>
                  <SurveySummaryResults />
                </Paper>
              </Box>
              <Box mb={3}>
                <Paper elevation={0}>
                  <SurveyAttachments />
                </Paper>
              </Box>
              {/* Temporarily hiding section while integrating */}
              {config?.CB_API_HOST ? (
                <Box mb={3}>
                  <Paper elevation={0}>
                    <SurveyAnimals />
                  </Paper>
                </Box>
              ) : null}
              <Box mb={3}>
                <Paper elevation={0}>
                  <SurveyStudyArea />
                </Paper>
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Container>
    </>
  );
};

export default SurveyPage;
