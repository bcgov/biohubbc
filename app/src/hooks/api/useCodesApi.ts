import { AxiosInstance } from 'axios';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';

/**
 * Returns a set of supported api methods for working with projects.
 *
 * @param {AxiosInstance} axios
 * @return {*} object whose properties are supported api methods.
 */
const useCodesApi = (axios: AxiosInstance) => {
  /**
   * Fetch all code sets.
   *
   * @return {*}  {Promise<IGetAllCodeSetsResponse>}
   */
  const getAllCodeSets = async (): Promise<IGetAllCodeSetsResponse> => {
    const { data } = await axios.get('/api/codes/');

    return data;
  };

  return {
    getAllCodeSets
  };
};

export default useCodesApi;
