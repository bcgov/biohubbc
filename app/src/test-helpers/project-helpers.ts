import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';

export const getProjectForViewResponse: IGetProjectForViewResponse = {
  id: 1,
  project: {
    project_name: 'Test Project Name',
    project_type: '1',
    start_date: '1998-10-10',
    end_date: '2021-02-26',
    project_activities: [1],
    completion_status: 'Active',
    publish_date: '2021-01-26'
  },
  permit: {
    permits: [
      {
        permit_number: '123',
        permit_type: 'Permit type'
      }
    ]
  },
  location: {
    location_description: 'Location description',
    geometry: []
  },
  objectives: {
    objectives: 'Et ad et in culpa si',
    caveats: 'sjwer bds'
  },
  coordinator: {
    first_name: 'Amanda',
    last_name: 'Christensen',
    email_address: 'amanda@christensen.com',
    coordinator_agency: 'Amanda and associates',
    share_contact_details: 'true'
  },
  iucn: {
    classificationDetails: [
      {
        classification: 'classification',
        subClassification1: 'sub classification 1',
        subClassification2: 'sub classification 2'
      },
      {
        classification: 'classification again',
        subClassification1: 'sub classification 1 again',
        subClassification2: 'sub classification 2 again'
      }
    ]
  },
  funding: {
    fundingSources: [
      {
        id: 0,
        agency_id: 1,
        agency_name: 'agency name',
        agency_project_id: 'ABC123',
        investment_action_category: 1,
        investment_action_category_name: 'investment action',
        funding_amount: 333,
        start_date: '2000-04-14',
        end_date: '2021-04-13',
        revision_count: 1
      }
    ]
  },
  partnerships: {
    indigenous_partnerships: ['partner 1', 'partner 2'],
    stakeholder_partnerships: ['partner 3', 'partner 4']
  }
};
