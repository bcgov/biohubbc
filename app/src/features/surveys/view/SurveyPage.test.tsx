import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { useBiohubApi } from 'hooks/useBioHubApi';
import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';
import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import React from 'react';
import { Router } from 'react-router';
import { getProjectForViewResponse } from 'test-helpers/project-helpers';
import { getSurveyForViewResponse } from 'test-helpers/survey-helpers';
import SurveyPage from './SurveyPage';
import { DialogContextProvider } from 'contexts/dialogContext';

const history = createMemoryHistory({ initialEntries: ['/projects/1/surveys/1'] });

jest.mock('../../../hooks/useBioHubApi');
const mockUseBiohubApi = {
  project: {
    getProjectForView: jest.fn<Promise<IGetProjectForViewResponse>, [number]>()
  },
  survey: {
    getSurveyForView: jest.fn<Promise<IGetSurveyForViewResponse>, [number]>(),
    deleteSurvey: jest.fn()
  },
  codes: {
    getAllCodeSets: jest.fn<Promise<IGetAllCodeSetsResponse>, []>()
  }
};

const mockBiohubApi = ((useBiohubApi as unknown) as jest.Mock<typeof mockUseBiohubApi>).mockReturnValue(
  mockUseBiohubApi
);

describe('SurveyPage', () => {
  beforeEach(() => {
    // clear mocks before each test
    mockBiohubApi().project.getProjectForView.mockClear();
    mockBiohubApi().survey.getSurveyForView.mockClear();
    mockBiohubApi().survey.deleteSurvey.mockClear();
    mockBiohubApi().codes.getAllCodeSets.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  const component = (
    <DialogContextProvider>
      <Router history={history}>
        <SurveyPage />
      </Router>
    </DialogContextProvider>
  );

  it('renders a spinner if no project is loaded', async () => {
    mockBiohubApi().survey.getSurveyForView.mockResolvedValue(getSurveyForViewResponse);
    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
      activity: [{ id: 1, name: 'activity 1' }]
    } as any);

    const { asFragment } = render(component);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders a spinner if no codes is loaded', async () => {
    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
    mockBiohubApi().survey.getSurveyForView.mockResolvedValue(getSurveyForViewResponse);

    const { asFragment } = render(component);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders a spinner if no survey is loaded', async () => {
    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
      activity: [{ id: 1, name: 'activity 1' }]
    } as any);

    const { asFragment } = render(component);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders survey page when survey is loaded', async () => {
    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
    mockBiohubApi().survey.getSurveyForView.mockResolvedValue(getSurveyForViewResponse);
    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
      activity: [{ id: 1, name: 'activity 1' }]
    } as any);

    const { asFragment, findByText } = render(component);

    const surveyHeaderText = await findByText('survey name', { selector: 'h1' });

    await waitFor(() => {
      expect(surveyHeaderText).toBeVisible();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders correctly with no end date', async () => {
    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
    mockBiohubApi().survey.getSurveyForView.mockResolvedValue({
      ...getSurveyForViewResponse,
      survey_details: {
        ...getSurveyForViewResponse.survey_details,
        end_date: (null as unknown) as string
      }
    });
    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
      activity: [{ id: 1, name: 'activity 1' }]
    } as any);

    const { asFragment, findByText } = render(component);

    const surveyHeaderText = await findByText('survey name', { selector: 'h1' });

    await waitFor(() => {
      expect(surveyHeaderText).toBeVisible();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('delete survey works and takes user to the surveys list page', async () => {
    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
    mockBiohubApi().survey.getSurveyForView.mockResolvedValue(getSurveyForViewResponse);
    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
      activity: [{ id: 1, name: 'activity 1' }]
    } as any);
    mockBiohubApi().survey.deleteSurvey.mockResolvedValue(true);

    const { getByTestId, getByText, findByText } = render(component);

    const surveyHeaderText = await findByText('survey name', { selector: 'h1' });

    await waitFor(() => {
      expect(surveyHeaderText).toBeVisible();
    });

    fireEvent.click(getByTestId('delete-survey-button'));

    await waitFor(() => {
      expect(
        getByText('Are you sure you want to delete this survey, its attachments and associated observations?')
      ).toBeInTheDocument();
    });

    fireEvent.click(getByTestId('yes-button'));

    await waitFor(() => {
      expect(history.location.pathname).toEqual(`/projects/${getProjectForViewResponse.id}/surveys`);
    });
  });
});
