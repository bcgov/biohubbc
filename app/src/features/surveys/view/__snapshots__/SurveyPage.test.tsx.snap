// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyPage renders a spinner if no codes is loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root pageProgress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SurveyPage renders a spinner if no project is loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root pageProgress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SurveyPage renders a spinner if no survey is loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root pageProgress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SurveyPage renders correctly with no end date 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation2"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiBox-root MuiBox-root-68"
      >
        <nav
          class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <ol
            class="MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                aria-current="page"
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-breadCrumbLink-61 MuiTypography-colorPrimary"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Projects
                </p>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                aria-current="page"
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-breadCrumbLink-61 MuiTypography-colorPrimary"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Test Project Name
                </p>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                survey name
              </p>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="MuiBox-root MuiBox-root-69"
      >
        <div
          class="MuiBox-root MuiBox-root-70"
        >
          <div
            class="MuiBox-root MuiBox-root-71"
          >
            <h1
              class="MuiTypography-root makeStyles-spacingRight-65 MuiTypography-h1"
            >
              survey name
            </h1>
            <div
              class="MuiChip-root makeStyles-chip-62 makeStyles-chipActive-63 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                ACTIVE
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-72"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              October 10, 1998
            </h6>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-73"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-actionButton-66 MuiButton-outlinedPrimary"
            data-testid="publish-survey-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,10A2,2 0 0,1 9,12A2,2 0 0,1 7,14A2,2 0 0,1 5,12A2,2 0 0,1 7,10M17,7A5,5 0 0,1 22,12A5,5 0 0,1 17,17H7A5,5 0 0,1 2,12A5,5 0 0,1 7,7H17M7,9A3,3 0 0,0 4,12A3,3 0 0,0 7,15H17A3,3 0 0,0 20,12A3,3 0 0,0 17,9H7Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Publish Survey
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-actionButton-66 MuiButton-outlinedPrimary"
            data-testid="delete-survey-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Delete Survey
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl"
  >
    <div
      class="MuiBox-root MuiBox-root-74"
    >
      <aside
        class="MuiBox-root MuiBox-root-75"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <nav
            aria-label="Survey Navigation"
            class="MuiList-root makeStyles-surveyNav-60 MuiList-padding"
            role="navigation"
          >
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/details"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Survey Details
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/results"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 7V21H18V23H4C2.9 23 2 22.1 2 21V7H4M12.8 15.35L9.5 12.05L10.9 10.65L12.8 12.55L17.1 8.25L18.5 9.65L12.8 15.35M20 3C21.1 3 22 3.9 22 5V17C22 18.1 21.1 19 20 19H8C6.9 19 6 18.1 6 17V5C6 3.9 6.9 3 8 3H11.18C11.6 1.84 12.7 1 14 1C15.3 1 16.4 1.84 16.82 3H20M14 3C13.45 3 13 3.45 13 4C13 4.55 13.45 5 14 5C14.55 5 15 4.55 15 4C15 3.45 14.55 3 14 3M10 7V5H8V17H20V5H18V7H10Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Summary Results
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/observations"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,9A3,3 0 0,0 9,12A3,3 0 0,0 12,15A3,3 0 0,0 15,12A3,3 0 0,0 12,9M12,17A5,5 0 0,1 7,12A5,5 0 0,1 12,7A5,5 0 0,1 17,12A5,5 0 0,1 12,17M12,4.5C7,4.5 2.73,7.61 1,12C2.73,16.39 7,19.5 12,19.5C17,19.5 21.27,16.39 23,12C21.27,7.61 17,4.5 12,4.5Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Observations
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/attachments"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5,6V17.5A4,4 0 0,1 12.5,21.5A4,4 0 0,1 8.5,17.5V5A2.5,2.5 0 0,1 11,2.5A2.5,2.5 0 0,1 13.5,5V15.5A1,1 0 0,1 12.5,16.5A1,1 0 0,1 11.5,15.5V6H10V15.5A2.5,2.5 0 0,0 12.5,18A2.5,2.5 0 0,0 15,15.5V5A4,4 0 0,0 11,1A4,4 0 0,0 7,5V17.5A5.5,5.5 0 0,0 12.5,23A5.5,5.5 0 0,0 18,17.5V6H16.5Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Attachments
                </span>
              </div>
            </a>
          </nav>
        </div>
      </aside>
      <article
        class="MuiBox-root MuiBox-root-76 makeStyles-article-67"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SurveyPage renders survey page when survey is loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation2"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiBox-root MuiBox-root-51"
      >
        <nav
          class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <ol
            class="MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                aria-current="page"
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-breadCrumbLink-44 MuiTypography-colorPrimary"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Projects
                </p>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                aria-current="page"
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-breadCrumbLink-44 MuiTypography-colorPrimary"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Test Project Name
                </p>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                survey name
              </p>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="MuiBox-root MuiBox-root-52"
      >
        <div
          class="MuiBox-root MuiBox-root-53"
        >
          <div
            class="MuiBox-root MuiBox-root-54"
          >
            <h1
              class="MuiTypography-root makeStyles-spacingRight-48 MuiTypography-h1"
            >
              survey name
            </h1>
            <div
              class="MuiChip-root makeStyles-chip-45 makeStyles-chipActive-46 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                ACTIVE
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-55"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              October 10, 1998 - February 26, 2021
            </h6>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-56"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-actionButton-49 MuiButton-outlinedPrimary"
            data-testid="publish-survey-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,10A2,2 0 0,1 9,12A2,2 0 0,1 7,14A2,2 0 0,1 5,12A2,2 0 0,1 7,10M17,7A5,5 0 0,1 22,12A5,5 0 0,1 17,17H7A5,5 0 0,1 2,12A5,5 0 0,1 7,7H17M7,9A3,3 0 0,0 4,12A3,3 0 0,0 7,15H17A3,3 0 0,0 20,12A3,3 0 0,0 17,9H7Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Publish Survey
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-actionButton-49 MuiButton-outlinedPrimary"
            data-testid="delete-survey-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Delete Survey
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl"
  >
    <div
      class="MuiBox-root MuiBox-root-57"
    >
      <aside
        class="MuiBox-root MuiBox-root-58"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <nav
            aria-label="Survey Navigation"
            class="MuiList-root makeStyles-surveyNav-43 MuiList-padding"
            role="navigation"
          >
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/details"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Survey Details
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/results"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 7V21H18V23H4C2.9 23 2 22.1 2 21V7H4M12.8 15.35L9.5 12.05L10.9 10.65L12.8 12.55L17.1 8.25L18.5 9.65L12.8 15.35M20 3C21.1 3 22 3.9 22 5V17C22 18.1 21.1 19 20 19H8C6.9 19 6 18.1 6 17V5C6 3.9 6.9 3 8 3H11.18C11.6 1.84 12.7 1 14 1C15.3 1 16.4 1.84 16.82 3H20M14 3C13.45 3 13 3.45 13 4C13 4.55 13.45 5 14 5C14.55 5 15 4.55 15 4C15 3.45 14.55 3 14 3M10 7V5H8V17H20V5H18V7H10Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Summary Results
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/observations"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,9A3,3 0 0,0 9,12A3,3 0 0,0 12,15A3,3 0 0,0 15,12A3,3 0 0,0 12,9M12,17A5,5 0 0,1 7,12A5,5 0 0,1 12,7A5,5 0 0,1 17,12A5,5 0 0,1 12,17M12,4.5C7,4.5 2.73,7.61 1,12C2.73,16.39 7,19.5 12,19.5C17,19.5 21.27,16.39 23,12C21.27,7.61 17,4.5 12,4.5Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Observations
                </span>
              </div>
            </a>
            <a
              class="MuiListItem-root MuiListItem-gutters"
              href="/admin/projects/1/surveys/attachments"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5,6V17.5A4,4 0 0,1 12.5,21.5A4,4 0 0,1 8.5,17.5V5A2.5,2.5 0 0,1 11,2.5A2.5,2.5 0 0,1 13.5,5V15.5A1,1 0 0,1 12.5,16.5A1,1 0 0,1 11.5,15.5V6H10V15.5A2.5,2.5 0 0,0 12.5,18A2.5,2.5 0 0,0 15,15.5V5A4,4 0 0,0 11,1A4,4 0 0,0 7,5V17.5A5.5,5.5 0 0,0 12.5,23A5.5,5.5 0 0,0 18,17.5V6H16.5Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Attachments
                </span>
              </div>
            </a>
          </nav>
        </div>
      </aside>
      <article
        class="MuiBox-root MuiBox-root-59 makeStyles-article-50"
      />
    </div>
  </div>
</DocumentFragment>
`;
